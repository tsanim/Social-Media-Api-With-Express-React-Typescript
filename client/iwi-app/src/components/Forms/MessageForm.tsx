import React from 'react';
import PropTypes from 'prop-types';
import MessageFormProps from '../../interfaces/Components/Forms/MessageFormProps.interface';

function MessageForm(props: MessageFormProps) {
    return (
        <div className="messageForm">
            <form id="messageForm">
                <input
                    type="text"
                    placeholder="Type a message"
                    name="messageText"
                    value={props.messageText}
                    onChange={props.onInputChangeHandler}
                    onKeyPress={props.onKeyPressHandler}
                    onKeyDown={props.onKeyDownHandler}
                    onBlur={props.onBlurHandler}
                />
                <input type="button" onClick={props.sendMessageHandler} value="Send" />
            </form>
        </div>
    )
}

MessageForm.propTypes = {
    messageText: PropTypes.string,
    onInputChangeHandler: PropTypes.func,
    onKeyPressHandler: PropTypes.func,
    onFocusHandler: PropTypes.func,
    onBlurHandler: PropTypes.func,
    sendMessageHandler: PropTypes.func,
    onKeyDownHandler: PropTypes.func,
}

export default MessageForm;
