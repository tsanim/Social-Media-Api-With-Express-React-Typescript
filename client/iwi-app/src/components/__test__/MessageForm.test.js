import React from 'react';
import { shallow } from 'enzyme';

import MessageForm from '../Forms/MessageForm';

function messageFormComponentSetup() {
    const props = {
        messageText: 'text',
        onInputChangeHandler: jest.fn(),
        onKeyPressHandler: jest.fn(),
        onFocusHandler: jest.fn(),
        onBlurHandler: jest.fn(),
        sendMessageHandler: jest.fn(),
        onKeyDownHandler: jest.fn()
    }

    const enzymeWrapper = shallow(<MessageForm {...props} />)

    return {
        props,
        enzymeWrapper
    }
}


describe('MessageForm Component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = messageFormComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render container div tag with class "messageForm"', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('div.messageForm')).toHaveLength(1);
    });

    it('should render container form tag with id "messageForm"', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('form#messageForm')).toHaveLength(1);
    });

    it('should render input with type "text"', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]')).toHaveLength(1);
    });

    it('should render input with type "button" ', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="button"]')).toHaveLength(1);
    });

    it('should render input type "text" with name "messageText"', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]').prop('name')).toEqual('messageText');
    });

    it('should render input with type "text" with onChange event', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]').prop('onChange')).toBeDefined();
    });

    it('should render input with type "text" with onKeyPress event', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]').prop('onKeyPress')).toBeDefined();
    });

    it('should render input with type "text" with onKeyDown event', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]').prop('onKeyDown')).toBeDefined();
    });

    it('should render input with type "text" with onBlur event', () => {
        const { enzymeWrapper } = messageFormComponentSetup();
        
        expect(enzymeWrapper.find('input[type="text"]').prop('onBlur')).toBeDefined();
    });

    it('should test if user text is echoed', () => {
        const { enzymeWrapper, props } = messageFormComponentSetup();

        expect(enzymeWrapper.find('input[type="text"]').props().value).toEqual(props.messageText);
    });
});