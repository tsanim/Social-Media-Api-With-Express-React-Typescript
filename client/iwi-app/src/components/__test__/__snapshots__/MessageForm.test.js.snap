// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm Component should render initial layout 1`] = `
Array [
  <div
    className="messageForm"
  >
    <form
      id="messageForm"
    >
      <input
        name="messageText"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        onKeyDown={[MockFunction]}
        onKeyPress={[MockFunction]}
        placeholder="Type a message"
        type="text"
        value="text"
      />
      <input
        onClick={[MockFunction]}
        type="button"
        value="Send"
      />
    </form>
  </div>,
]
`;
